### PR TITLE
Fix prop validation diagnostics not working for PascalCase components in templates 

### PR DESCRIPTION
**Pull request by triforcely**
This PR fixes problem with 'PascalCase' component names in vue templates.

### CODE REVIEW

**Comment by triforcely** [comment]
https://github.com/vuejs/vetur/issues/2168

File: server/src/modes/template/services/vuePropValidation.ts [hunk 1]
```diff
@@ -14,10 +14,15 @@ export function doPropValidation(document: TextDocument, htmlDocument: HTMLDocum
   });
 
   traverseNodes(htmlDocument.roots, n => {
-    if (n.tag && Object.keys(childComponentToProps).includes(n.tag)) {
-      const d = generateDiagnostic(n, childComponentToProps[n.tag], document);
-      if (d) {
-        diagnostics.push(d);
+    if (n.tag) {
+      const kebabCaseTag = kebabCase(n.tag);
```
Comment (yoyo930021) [inline server/src/modes/template/services/vuePropValidation.ts:18]: Maybe we can use `getSameTagInSet` function?

```typescript
const componetProps = getSameTagInSet(childComponentToProps, n.tag)
if (componetProps) {
  const d = generateDiagnostic(n, componetProps, document);
  if (d) {
    diagnostics.push(d);
  }
}
```

Comment (triforcely) [reply server/src/modes/template/services/vuePropValidation.ts]: Hey. I'm new to the code base, so I didn't know this utility function exists. Thank you, I updated the code!

Comment (yoyo930021) [reply server/src/modes/template/services/vuePropValidation.ts]: It's great.

**Comment by triforcely** [comment]
Does anything else need to be added to this PR for it to get merged?

**Comment by yoyo930021** [comment]
> Does anything else need to be added to this PR for it to get merged?

You only need to wait for review by @octref.

**Review by octref — APPROVED** [review]
Good work 👍 
Only need to fix the Changelog so I made the change myself. Hope you don't mind.

**Pull request merged**